src/main/java/org/broadinstitute/hellbender/utils/IndexUtils.java

```diff
@@ -0,0 +1,109 @@
+import htsjdk.samtools.SAMSequenceRecord;
+import htsjdk.tribble.AbstractFeatureReader;
+import htsjdk.tribble.Tribble;
+import htsjdk.tribble.index.Index;
+import htsjdk.tribble.index.IndexFactory;
+import htsjdk.tribble.util.ParsingUtils;
+import htsjdk.tribble.util.TabixUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.IndexFeatureFile;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+public final class IndexUtils {
+    private IndexUtils(){}
+
+    private static final Logger logger = LogManager.getLogger(IndexUtils.class);
+
+    /**
+     * Load the index from disk, checking for out of date indexes and old versions
+     * @return an Index, or null if we're unable to load
+     */
+    public static Index loadTribbleIndex(final File featureFile) {
+        Utils.nonNull(featureFile);
+        final File indexFile = Tribble.indexFile(featureFile);
+        if (! indexFile.canRead()) {
+            return null;
+        }
+        logger.debug("Loading Tribble index from disk for file " + featureFile);
+        final Index index = IndexFactory.loadIndex(indexFile.getAbsolutePath());
+        checkIndexModificationTime(featureFile, indexFile, index);
+        return index;
+    }
+
+    /**
+     * Try to load the tabix index from disk, checking for out of date indexes and old versions
+     * @return an Index, or null if we're unable to load
+     */
+    public static Index loadTabixIndex(final File featureFile) {
+        try {
+            final String path = featureFile.getAbsolutePath();
+            final boolean isTabix = new AbstractFeatureReader.ComponentMethods().isTabix(path, null);
+            if (! isTabix){
+                return null;
+            }
+            final String indexPath = ParsingUtils.appendToPath(path, TabixUtils.STANDARD_INDEX_EXTENSION);
+            logger.debug("Loading tabix index from disk for file " + featureFile);
+            final Index index = IndexFactory.loadIndex(indexPath);
+            final File indexFile = new File(indexPath);
+            checkIndexModificationTime(featureFile, indexFile, index);
+            return index;
+        } catch (final IOException e) {
+            return null;
+        }
+    }
+
+    /**
+     * @throws UserException the index is not the current version.
+     * Prints a warning if the index is not up-to date (ie older than the feature file)
+     */
+    public static void checkIndexModificationTime(final File featureFile, final File indexFile, final Index index) {
+        Utils.nonNull(featureFile, "featureFile");
+        Utils.nonNull(indexFile, "indexFile");
+        Utils.nonNull(index, "index");
+        if (! index.isCurrentVersion()) {
+            // we've loaded an old version of the index, we want to remove it <-- currently not used, but may re-enable
```

Comment:
Remove obsolete "<-- currently not used, but may re-enable" comment inherited from GATK3.